### PR TITLE
Feature/160 161 sql plugins(sql server db flags)

### PR DESCRIPTION
**Pull request by umermehmood**

### CODE REVIEW

File: plugins/google/sql/sqlCrossDbOwnership.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+var async = require('async');
+var helpers = require('../../../helpers/google');
+
+module.exports = {
+    title: 'SQLServer Cross Db Ownership Chaining',
+    category: 'SQL',
+    description: 'Ensures SQL instances for SQLServer type have cross db ownership chaining flag disabled.',
+    more_info: 'SQL instance for SQLServer databases provides cross db ownership chaining flag. It is used to configure cross database ownership chaining for all databases. It is enabled by default. We should disabled it for security unless all databases agrees for this setting.',
```
Comment (AkhtarAmir) [inline plugins/google/sql/sqlCrossDbOwnership.js:8]:
```suggestion
    more_info: 'SQL instance for SQLServer databases provides cross DB ownership chaining flag. It is used to configure cross-database ownership chaining for all databases. It is enabled by default and should be disabled for security unless all databases agree for this setting.',
```

File: plugins/google/sql/sqlCrossDbOwnership.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    link: 'https://cloud.google.com/sql/docs/sqlserver/flags',
+    recommended_action: 'Ensure that cross db ownership chaining flag is disabled for all SQLServer instances.',
```
Comment (AkhtarAmir) [inline plugins/google/sql/sqlCrossDbOwnership.js:10]:
```suggestion
    recommended_action: 'Ensure that cross DB ownership chaining flag is disabled for all SQLServer instances.',
```

File: plugins/google/sql/sqlCrossDbOwnership.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                if (sqlInstance.instanceType != "READ_REPLICA_INSTANCE" &&
+                    sqlInstance.settings &&
+                    sqlInstance.settings.databaseFlags &&
+                    sqlInstance.settings.databaseFlags.length) {
+                        let found = sqlInstance.settings.databaseFlags.find(flag => flag.name && flag.name == 'cross db ownership chaining' &&
+                                                                        flag.value && flag.value == 'off');
+
+                        if (found) {
+                            helpers.addResult(results, 0, 
+                                'SQL instance does not have cross db ownership chaining flag enabled', region, sqlInstance.name);
```
Comment (AkhtarAmir) [inline plugins/google/sql/sqlCrossDbOwnership.js:50]:
```suggestion
                                'SQL instance does not have cross DB ownership chaining flag enabled', region, sqlInstance.name);
```

File: plugins/google/sql/sqlCrossDbOwnership.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+                        } else {
+                            helpers.addResult(results, 2,
+                                'SQL instance have cross db ownership chaining flag enabled', region, sqlInstance.name);
```
Comment (AkhtarAmir) [inline plugins/google/sql/sqlCrossDbOwnership.js:53]:
```suggestion
                                'SQL instance has cross DB ownership chaining flag enabled', region, sqlInstance.name);
```

File: plugins/google/sql/sqlCrossDbOwnership.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+                        }
+                } else if (sqlInstance.instanceType == "READ_REPLICA_INSTANCE") {
+                } else {
+                    helpers.addResult(results, 2, 
+                        'SQL instance have cross db ownership chaining flag enabled', region, sqlInstance.name);
```
Comment (AkhtarAmir) [inline plugins/google/sql/sqlCrossDbOwnership.js:58]:
```suggestion
                        'SQL instance has cross DB ownership chaining flag enabled', region, sqlInstance.name);
```

File: plugins/google/sql/sqlContainedDatabaseAuth.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
+var async = require('async');
+var helpers = require('../../../helpers/google');
+
+module.exports = {
+    title: 'SQLServer Contained Database Authentication',
+    category: 'SQL',
+    description: 'Ensures SQL instances for SQLServer type have Contained Database Authentication flag disabled.',
+    more_info: 'SQL instance for SQLServer databases provides Contained Database Authentication flag. It is used to shift the user authentication prcoess from database engine level to database level. It is enabled by default. We should always disable it to keep the authentication enabled for security.',
```
Comment (AkhtarAmir) [inline plugins/google/sql/sqlContainedDatabaseAuth.js:8]:
```suggestion
    more_info: 'SQL instance for SQLServer databases provides Contained Database Authentication flag. It is used to shift the user authentication process from database engine level to database level. It is enabled by default and should always be disabled to keep the authentication enabled for security.',
```

File: plugins/google/sql/sqlContainedDatabaseAuth.js [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+                            helpers.addResult(results, 0, 
+                                'SQL instance does not have contained database authentication flag enabled', region, sqlInstance.name);
+                        } else {
+                            helpers.addResult(results, 2,
+                                'SQL instance have contained database authentication flag enabled', region, sqlInstance.name);
+                        }
+                } else if (sqlInstance.instanceType == "READ_REPLICA_INSTANCE") {
+                } else {
+                    helpers.addResult(results, 2, 
+                        'SQL instance have contained database authentication flag enabled', region, sqlInstance.name);
```
Comment (AkhtarAmir) [inline plugins/google/sql/sqlContainedDatabaseAuth.js:58]:
```suggestion
                        'SQL instance has contained database authentication flag enabled', region, sqlInstance.name);
```

File: plugins/google/sql/sqlContainedDatabaseAuth.js [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+                    sqlInstance.settings.databaseFlags.length) {
+                        let found = sqlInstance.settings.databaseFlags.find(flag => flag.name && flag.name == 'contained database authentication' &&
+                                                                        flag.value && flag.value == 'off');
+
+                        if (found) {
+                            helpers.addResult(results, 0, 
+                                'SQL instance does not have contained database authentication flag enabled', region, sqlInstance.name);
+                        } else {
+                            helpers.addResult(results, 2,
+                                'SQL instance have contained database authentication flag enabled', region, sqlInstance.name);
```
Comment (AkhtarAmir) [inline plugins/google/sql/sqlContainedDatabaseAuth.js:53]:
```suggestion
                                'SQL instance has contained database authentication flag enabled', region, sqlInstance.name);
```

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

File: plugins/google/sql/sqlContainedDatabaseAuth.js [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+                            helpers.addResult(results, 0, 
+                                'SQL instance does not have contained database authentication flag enabled', region, sqlInstance.name);
+                        } else {
+                            helpers.addResult(results, 2,
+                                'SQL instance has contained database authentication flag enabled', region, sqlInstance.name);
+                        }
+                } else if (sqlInstance.instanceType == "READ_REPLICA_INSTANCE") {
+                } else {
+                    helpers.addResult(results, 2, 
+                        'SQL instance has contained database authentication flag enabled', region, sqlInstance.name);
```
Comment (giorod3) [inline plugins/google/sql/sqlContainedDatabaseAuth.js:58]: this should pass if there are no db flags

File: plugins/google/sql/sqlCrossDbOwnership.js [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+                        if (found) {
+                            helpers.addResult(results, 0, 
+                                'SQL instance does not have cross DB ownership chaining flag enabled', region, sqlInstance.name);
+                        } else {
+                            helpers.addResult(results, 2,
+                                'SQL instance has cross DB ownership chaining flag enabled', region, sqlInstance.name);
+                        }
+                } else if (sqlInstance.instanceType == "READ_REPLICA_INSTANCE") {
+                } else {
+                    helpers.addResult(results, 2,
```
Comment (giorod3) [inline plugins/google/sql/sqlCrossDbOwnership.js:57]: same here, this should pass

File: plugins/google/sql/sqlContainedDatabaseAuth.js (hunk 9 above)
Comment (umermehmood) [reply plugins/google/sql/sqlContainedDatabaseAuth.js]: Updated.

File: plugins/google/sql/sqlCrossDbOwnership.js (hunk 10 above)
Comment (umermehmood) [reply plugins/google/sql/sqlCrossDbOwnership.js]: Updated

**Review by AkhtarAmir — APPROVED** [review]

**Review by giorod3 — APPROVED** [review]

**Pull request merged**